lib/values.rb

```diff
@@ -23,17 +23,23 @@ def self.new(*fields, &block)
     Class.new do
       attr_reader(:hash, *fields)
 
-      define_method(:initialize) do |*values|
-        raise ArgumentError.new("wrong number of arguments, #{values.size} for #{fields.size}") if fields.size != values.size
+      instance_var_assignments = Array.new(fields.length) do |idx|
+        "@#{fields[idx]} = values[#{idx}]"
+      end.join("\n")
+
+      class_eval <<-RUBY
```

Comment:
this looks great, thanks! Mind adding a comment here mentioning this optimization? Whilst this library doesn't change much, I like to prevent opportunities for bitrot where possible.